crates/ty_python_semantic/src/types.rs

```diff
@@ -8294,6 +8310,25 @@ impl<'db> ModuleLiteralType<'db> {
         Some(Type::module_literal(db, importing_file, submodule))
     }
 
+    fn try_module_getattr(self, db: &'db dyn Db, name: &str) -> PlaceAndQualifiers<'db> {
+        // For module literals, we want to try calling the module's own __getattr__ function
+        // if it exists. First, we need to look up the __getattr__ function in the module's scope.
+        if let Some(file) = self.module(db).file(db) {
+            let getattr_symbol = imported_symbol(db, file, "__getattr__", None);
+            if let Place::Type(getattr_type, _) = getattr_symbol.place {
+                // If we found a __getattr__ function, try to call it with the name argument
+                if let Ok(outcome) = getattr_type.try_call(
```

Comment:
I think it's correct that you didn't actually reuse the existing call-getattr logic here, because it doesn't really apply. E.g. type-level `__getattr__` has to be loaded from the meta-type (as with any other dunder method), so we use `try_call_dunder` -- but that doesn't apply to module-level getattr. The logic you have here looks correct to me.